cylc/flow/config.py

```diff
@@ -789,12 +789,21 @@ def process_start_cycle_point(self):
             self.start_point
         """
         if getattr(self.options, 'startcp', None) is not None:
-            # Warm start from a point later than initial point.
+            # Start from a point later than initial point.
             if self.options.startcp == 'now':
                 self.options.startcp = get_current_time_string()
             self.start_point = get_point(self.options.startcp).standardise()
+        elif getattr(self.options, 'starttask', None) is not None:
+            # Start from designated task(s).
+            # Select the earliest start point for use in pre-initial ignore.
+            self.start_point = min(
+                get_point(
+                    TaskID.split(taskid)[1]
+                ).standardise()
+                for taskid in self.options.starttask
+            )
```

Comment:
The only thing I had to do for this PR was define the start cycle point, for the start task case. Then pre-initial ignore gets done automatically.

Reply:
Hmm.. Is this the proper way of doing it?

What if the workflow has some R1 / start-up tasks for a few cycles, and then runs it's normal tasks (containing the start task) at some offset from ICP?

Reply:
Sorry, I don't follow here either :thinking:  If we start up the scheduler with `--start-task` all that matters (for the purpose of this PR) is that other tasks don't get stuck waiting on tasks at cycle points earlier than the start-task(s).  How do R1 tasks and offsets from ICP affect that?

Reply:
The start task (and perhaps dependents) may not exist at the ICP.  They could be defined as starting from some offset of the ICP...

So if we force the ICP to a non-ICP start task.. What is the result?
(might test this tomorrow)

Reply:
You have to give the name.point of start-tasks, and they are checked for validity at the specified point.
```ini
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    [[graph]]
        R1 = prep => foo
        P1 = foo[-P1] => foo
[runtime]
    [[prep, foo]]
```
 
![shot](https://user-images.githubusercontent.com/2362137/124434738-57225e80-ddc8-11eb-983b-c42b14e019a2.png)

Reply:
Hmm, maybe my statement above wasn't clear:
> The only thing I had to do for this PR was define the start cycle point, for the start task case. Then pre-initial ignore gets done automatically.

Rather than *define the start cycle point for start tasks* I should have said, *extract the cycle point from start tasks and set the workflow start point to that*.

Reply:
What I'm asking is; how do you square the making the workflow starting point the with the task starting point when the task is defined in the workflow graph to be some offset from the starting point?
```
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    [[graph]]
        R1 = prep => foo
        +P5/P1 = foo[-P1] => foo => bar
[runtime]
    [[prep, foo, bar]]
```
And then run with:
`$ cylc play --start-task=bar.10 demo2/run1`

Reply:
by start point I assume you mean `initial cycle point` at run time (?).

Reply:
Right, understood now. No, ICP defines the extent of the graph (and where things happen relative to ICP) but *start point* (or start task now too) is where you start up within that graph. So we don’t mess with ICP here.

Reply:
Thanks, well, apart from the minor conflict.. wer're ready to go with this one.